File: July2022/C++/FibonacciNumber.cpp
class Solution {
public:
    int fib(int n) {
        
        int a = 0, b = 1, c = 0;
        for(int i=0;i<n;i++)
        {
            a = b;
            b = c;
            c = a + b;
        }
        return c;
    }
};

File: July2022/Java/FibonacciNumber.java

// @saorav21994
// TC : O(n)
// SC : O(1)

class Solution {
    public int fib(int n) {
        int res = 0;
        int a = 0, b = 1;
        if (n == 1) return 1;
        for (int i = 2; i <= n; i++) {
            res = a + b;
            a = b;
            b = res;
        }
        return res;
    }
}
